### PR TITLE
Make VarBitSet saturated + FromIterator efficient

### PR DESCRIPTION
**Pull request by sameer**
I noticed that `VarBitSet::saturated` takes a significant amount of time when generating data with `VecStrategy`. There are two perf issues:

1. not making use of the size hint in `FromIterator` leads to repeated grow calls
2. saturated should use `from_elem` since we're setting all bits

### CODE REVIEW

**Comment by sameer** [comment]
I'm using [Instruments](https://help.apple.com/instruments/mac/10.0/#/dev7b09c84f5) on macOS to look at the CPU profile of a proptest and it's now optimized to `memset` 🚀 

Before:
Screenshot 2023-08-17 at 13 00 02

After:

Screenshot 2023-08-17 at 13 01 35

**Review by matthew-russo — APPROVED** [review]
Nice digging! This is a great find

**Comment by sameer** [comment]
@matthew-russo are you able to approve the workflows / merge this?

**Comment by matthew-russo** [comment]
I didn't realize the workflows weren't running. just kicked that off. we normally try to get two maintainers eyes on things before merging -- i pinged the others yesterday to give a look.

**Review by rex-remind101 — APPROVED** [review]
very nice 👍

**Pull request merged**
